src/features/HomePage/MoviesGallery/MovieCard/MovieCard.jsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faThumbsUp, faThumbsDown } from '@fortawesome/free-solid-svg-icons';
+
+import Rating from '../../../../components/Rating/Rating';
+import { addLikeAsync, changeRatingAsync } from '../../../../redux/movies/movies.async.actions';
+import { propTypesShapes } from '../../../../constants';
+import styles from './MovieCard.module.scss';
+
+const MovieCard = ({ movieData, addLikeAsync, changeRatingAsync }) => {
+  const { id, title, posterUrl, likes, stars } = movieData;
+
+  const changeLikesHandler = (likesAmount) => {
+    const updatedData = {
+      ...movieData,
+      likes: movieData.likes + likesAmount
+    };
+
+    addLikeAsync(id, updatedData);
+  };
+
+  const changeRatingHandler = (movieId, newRating) => {
+    const updatedData = {
+      ...movieData,
+      stars: newRating
+    };
+
+    changeRatingAsync(id, updatedData, newRating)
+  };
```

Comment:
1. You can use `useCallback` to memoize func here, it will optimize performance
2. `changeRatingAsync`, `addLikeAsync ` is async functions, but you don't use await with them.